### PR TITLE
Allow nil to WithError

### PR DESCRIPTION
**Pull request by vasyharan**

### CODE REVIEW

File: slog/slog_test.go [hunk 1]
```diff
@@ -348,6 +348,17 @@ func TestError(t *testing.T) {
 	if e.Err != errA.Error() {
 		t.Errorf("error not included\ngot: %v", e.Err)
 	}
+	e = e.WithError(nil)
+	e.Info("testing")
+	got = buf.String()
+	buf.Reset()
+	if strings.Contains(got, "error") {
+		t.Errorf("error persist when it shouldn't\ngot: %v", got)
```
Comment (IDispose) [inline slog/slog_test.go:356]: nitpick `error persists ...`

File: slog/slog_test.go [hunk 2]
```diff
@@ -358,19 +369,42 @@ func TestError(t *testing.T) {
 	if e.Err == errA.Error() {
 		t.Errorf("error not included\ngot: %v", e.Err)
 	}
+	e = e.WithError(nil)
+	e.Info("testing")
+	got = buf.String()
+	buf.Reset()
+	if strings.Contains(got, "error") {
+		t.Errorf("error persist when it shouldn't\ngot: %v", got)
```
Comment (IDispose) [inline slog/slog_test.go:377]: same nitpick here :-)

**Review by IDispose — APPROVED** [review]
minor nitpicks

File: slog/slog_test.go (hunk 2 above)
Comment (vasyharan) [reply slog/slog_test.go]: done

**Pull request merged**